File: 0014-longest-common-prefix/0014-longest-common-prefix.java
class Solution {
    public String maxCommon(String ans, String st) {
        int maxL = 0;
        int count = 0;
        int n = Math.min(ans.length(), st.length());
        //if(ans.isEmpty()) return st;
        //if(st.isEmpty()) return ans;
        for(int i=0; i<n; i++) {
            if(ans.charAt(i) == st.charAt(i)) {
                count++;
            }
            else break;
        }
        maxL = Math.max(maxL, count);
        if(maxL==0 && count==0) return "";
        if(ans.length() > st.length()) return ans.substring(0, maxL).trim();
        else return st.substring(0, maxL).trim();
    }
    public String longestCommonPrefix(String[] strs) {
        String ans = strs[0];
        if(strs.length == 0) return ans;
        for(int i=1; i<strs.length; i++) {
            ans = maxCommon(ans, strs[i]);
        }
        return ans;
    }
}

File: 0014-longest-common-prefix/0014-longest-common-prefix.py
class Solution:
    def maxCommon(self, ans, st):
        maxL = 0
        count = 0
        n = min(len(ans), len(st))

        for i in range(n) :
            if ans[i] == st[i]:
                count += 1
            else: break
        maxL = max(maxL, count)
        if len(ans)>len(st): return ans[0: maxL]
        else: return st[0:maxL]
    def longestCommonPrefix(self, strs: List[str]) -> str:
        ans = strs[0]
        if(len(strs) == 0):
            return ans

        for i in range(1,len(strs)):
            ans = self.maxCommon(ans, strs[i])
        return ans